pkg/minikube/bootstrapper/bsutil/kverify/node_health.go

```diff
@@ -0,0 +1,145 @@
+// NodeConditions verfies that node is not under disk, memory, pid or network pressure.
+func NodeConditions(cs *kubernetes.Clientset, drver string) error {
+	glog.Info("verifying NodePressure condition ...")
+	start := time.Now()
+	defer func() {
+		glog.Infof("duration metric: took %s to wait for NodePressure...", time.Since(start))
+	}()
+
+	ns, err := cs.CoreV1().Nodes().List(meta.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "list nodes")
+	}
+
+	for _, n := range ns.Items {
+		glog.Infof("node storage ephemeral capacity is %s", n.Status.Capacity.StorageEphemeral())
+		glog.Infof("node cpu capacity is %s", n.Status.Capacity.Cpu().AsDec())
+		for _, c := range n.Status.Conditions {
+			if c.Type == v1.NodeDiskPressure && c.Status == v1.ConditionTrue {
+				out.Ln("")
+				out.ErrT(out.FailureType, "node {{.name}} has unwanted condition {{.condition_type}} : Reason {{.reason}} Message: {{.message}}", out.V{"name": n.Name, "condition_type": c.Type, "reason": c.Reason, "message": c.Message})
+				out.WarningT("The node on {{.name}} has ran out of disk space. please consider allocating more disk using or pruning un-used images", out.V{"name": n.Name})
+				if driver.IsKIC(drver) && runtime.GOOS != "linux" {
+					out.T(out.Stopped, "Please increase Docker Desktop's disk image size.")
+					if runtime.GOOS == "darwin" {
+						out.T(out.Documentation, "Documentation: {{.url}}", out.V{"url": "https://docs.docker.com/docker-for-mac/space/"})
+					}
+					if runtime.GOOS == "windows" {
+						out.T(out.Documentation, "Documentation: {{.url}}", out.V{"url": "https://docs.docker.com/docker-for-windows/"})
+					}
+				} else { // VM-drivers
+					out.T(out.Stopped, "You can specify a larger disk for your cluster using `minikube start --disk` ")
+				}
+				out.Ln("") // if there is error message, lets make an empty space for better visilibtly
+				return fmt.Errorf("node %q has unwanted condition %q : Reason %q Message: %q ", n.Name, c.Type, c.Reason, c.Message)
+			}
+
+			if c.Type == v1.NodeMemoryPressure && c.Status == v1.ConditionTrue {
+				out.Ln("")
+				out.ErrT(out.FailureType, "node {{.name}} has unwanted condition {{.condition_type}} : Reason {{.reason}} Message: {{.message}}", out.V{"name": n.Name, "condition_type": c.Type, "reason": c.Reason, "message": c.Message})
+				out.WarningT("The node on {{.name}} has ran of memory.", out.V{"name": n.Name})
+				if driver.IsKIC(drver) && runtime.GOOS != "linux" {
+					out.T(out.Stopped, "Please increase Docker Desktop's memory.")
+					if runtime.GOOS == "darwin" {
+						out.T(out.Documentation, "Documentation: {{.url}}", out.V{"url": "https://docs.docker.com/docker-for-mac/space/"})
+					}
+					if runtime.GOOS == "windows" {
+						out.T(out.Documentation, "Documentation: {{.url}}", out.V{"url": "https://docs.docker.com/docker-for-windows/"})
+					}
+				} else {
+					out.T(out.Stopped, "You can specify a larger memory size for your cluster using `minikube start --memory` ")
```

Comment:
This advice does not cover Docker on Linux, none, or Podman.

Reply:
for docker on linux there is nothing they can do. they would need to buy a computer with larger memory. what should be the output ?